### PR TITLE
Design doc: scikit-learn-dask thoughts

### PR DESCRIPTION
**Pull request by mrocklin**
We shouldn't merge this.  These are just some loose thoughts.  Comments welcome.

@ogrisel @amueller @cowlicks

### CODE REVIEW

File: docs/source/learn-prototype.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+We create the following class hierarchy
+
+    Base - DaskEstimator - ParameterSearch - GridSearch
+                         \
+                          - Pipeline
+
+DaskEstimator supports the following subset of the scikit-learn API
+
+    clone
+    fit
```
Comment (amueller) [inline docs/source/learn-prototype.md]: It's likely that fit can defer any actual computation until predict or score is called.

Comment (mrocklin) [inline docs/source/learn-prototype.md]: Good point

**Comment by mrocklin** [comment]
@amueller and I have a working prototype for pipeline.

**Comment by mrocklin** [comment]
Prototype lives here https://github.com/mrocklin/dask-learn

**Comment by mrocklin** [comment]
Closing in favor of https://github.com/blaze/dask/pull/743

**Pull request closed without merging**